bridgenode/flatfileworker.go

```diff
@@ -312,23 +325,45 @@ func (ff *flatFileState) writeProofBlock(ud btcacc.UData) error {
 
 func (ff *flatFileState) writeTTLs(ttlRes ttlResultBlock) error {
 	var ttlArr [4]byte
+	var buffer [8]byte
+
+	// write offset of current TTL to offset file
+	ff.offsets = append(ff.offsets, ff.currentOffset)
+
+	// write ttl offset to offsetfile
+	binary.BigEndian.PutUint64(buffer[:], uint64(ff.currentOffset))
+	_, err := ff.offsetFile.WriteAt(buffer[:], int64(8*ttlRes.Height))
+	if err != nil {
+		return err
+	}
+
 	// for all the TTLs, seek and overwrite the empty values there
 	for _, c := range ttlRes.Created {
 		// seek to the location of that txo's ttl value in the proof file
 
 		binary.BigEndian.PutUint32(
 			ttlArr[:], uint32(ttlRes.Height-c.createHeight))
+		// for each iteration of  append 0x00
+		_, err := ff.proofFile.WriteAt([]byte{0x00, 0x00, 0x00, 0x00}, ff.currentOffset)
+		if err != nil {
+			return err
+		}
+
+		// increment value of offset after before write operation by 4
+		ff.currentOffset = ff.currentOffset + 4
```

Comment:
There's no reason to do this every loop. We can do

```
ff.currentOffset += len(ttlRes.Created) * 4
```

at the end of the loop.

Reply:
Okay Ill change it